packages/kafka-iam-auth/src/index.ts

```diff
@@ -230,17 +249,25 @@ const initKafka = (config: InteropKafkaConfig): Kafka => {
   });
 };
 
-const initConsumer = async (
-  config: KafkaConsumerConfig,
-  topics: string[],
-  consumerHandler: (payload: EachMessagePayload) => Promise<void>
-): Promise<Consumer> => {
+const initCustomConsumer = async ({
```

Comment:
here's a refactor suggestion, let me know what you think
(right now I do not have better ideas to improve configs parameters)

```
const initCustomConsumer = async ({
  config,
  topics,
  consumerRunConfig,
  batchConfig,
}: {
  config: KafkaConsumerConfig;
  topics: string[];
  consumerRunConfig: (consumer: Consumer) => ConsumerRunConfig
  batchConfig?: {minBytes : number, maxWaitTimeInMs: number},
} ): Promise<Consumer> => {
  genericLogger.debug(
    `Consumer connecting to topics ${JSON.stringify(topics)}`
  );

  const kafka = initKafka(config);

  const consumer = kafka.consumer({
    groupId: config.kafkaGroupId,
    retry: {
      initialRetryTime: 100,
      maxRetryTime: 3000,
      retries: 3,
      restartOnFailure: (error) => {
        genericLogger.error(`Error during restart service: ${error.message}`);
        return Promise.resolve(false);
      },
    },
    ...batchConfig,
  });

  if (config.resetConsumerOffsets) {
    await resetPartitionsOffsets(topics, kafka, consumer);
  }

  consumerKafkaEventsListener(consumer);
  errorEventsListener(consumer);

  await consumer.connect();
  genericLogger.debug("Consumer connected");

  const topicExists = await validateTopicMetadata(kafka, topics);
  if (!topicExists) {
    processExit();
  }

  await consumer.subscribe({
    topics,
    fromBeginning: config.topicStartingOffset === "earliest",
  });

  genericLogger.info(`Consumer subscribed topic ${topics}`);

  await consumer.run(consumerRunConfig(consumer))
  return consumer;
};
```

Reply:
I know, I encountered issues in finding a linear solution. Anyways, updated to your suggestion.